### ISSUE TITLE
Unable to authenticate - `curl_post` in `authenticate` returns false

### ISSUE THREAD

**Issue by stooj** [BrandonDusseau/np2-wallboard#12]
Thanks for making this! It looks brilliant!

Running Ubuntu 20.04 with PHP 8.

```
[Mon Sep 20 19:04:18.276936 2021] [php:notice] [pid 33519] [client xxx.xxx.xxx.xxx:38510] PHP Notice:   in /var/www/np/vendor/brandondusseau/phptriton/util.php on line 20
[Mon Sep 20 19:04:18.277268 2021] [php:warn] [pid 33519] [client xxx.xxx.xxx.xxx:38510] PHP Warning:  Undefined array key 1 in /var/www/np/vendor/brandondusseau/phptriton/util.php on line 32
[Mon Sep 20 19:04:18.277423 2021] [php:warn] [pid 33519] [client xxx.xxx.xxx.xxx:38510] PHP Warning:  Undefined array key 2 in /var/www/np/vendor/brandondusseau/phptriton/util.php on line 52
[Mon Sep 20 19:04:18.277535 2021] [php:warn] [pid 33519] [client xxx.xxx.xxx.xxx:38510] PHP Warning:  Trying to access array offset on value of type null in /var/www/np/vendor/brandondusseau/phptriton/client.php on line 42
[Mon Sep 20 19:04:18.277701 2021] [php:warn] [pid 33519] [client xxx.xxx.xxx.xxx:38510] PHP Warning:  Trying to access array offset on value of type null in /var/www/np/vendor/brandondusseau/phptriton/client.php on line 47
```

Let me know if you need any more details.

**Comment by BrandonDusseau** [comment]
Interesting. I haven't encountered such an issue myself. I will try running this in a PHP 8 Docker container to see if I am able to reproduce the issue. It may be a couple days, but I will definitely look into this.

By the way, you might want to redact your public IP address from the logs you shared.

**Comment by stooj** [comment]
> It may be a couple days, but I will definitely look into this.

Great news, thanks.

Redacted too, thanks for the heads up.

**Comment by BrandonDusseau** [comment]
There was an issue with PHP 8 having dropped support for the style of class constructor that the phpTriton library was using. That's now been fixed and you can expect v1.2.0 to be released within a few minutes.

**Comment by stooj** [comment]
Thanks so much for this, Brandon, and the quick turn arourd. I've tested the update and it is looking grand.
